src/libraries/System.Private.CoreLib/src/System/Enum.cs

```diff
@@ -345,43 +345,83 @@ public static bool IsDefined(Type enumType, object value)
+        {
+            bool success = TryParse(enumType, value, ignoreCase, throwOnFailure: true, out object? result);
+            Debug.Assert(success);
+            return result!;
+        }
+
         public static TEnum Parse<TEnum>(string value) where TEnum : struct =>
             Parse<TEnum>(value, ignoreCase: false);
 
+        public static TEnum Parse<TEnum>(ReadOnlySpan<char> value) where TEnum : struct =>
+           Parse<TEnum>(value, ignoreCase: false);
+
         public static TEnum Parse<TEnum>(string value, bool ignoreCase) where TEnum : struct
         {
             bool success = TryParse<TEnum>(value, ignoreCase, throwOnFailure: true, out TEnum result);
             Debug.Assert(success);
             return result;
         }
 
+        public static TEnum Parse<TEnum>(ReadOnlySpan<char> value, bool ignoreCase) where TEnum : struct
+        {
+            bool success = TryParse<TEnum>(value, ignoreCase, throwOnFailure: true, out TEnum result);
+            Debug.Assert(success);
+            return result;
+        }
+
         public static bool TryParse(Type enumType, string? value, out object? result) =>
             TryParse(enumType, value, ignoreCase: false, out result);
 
+        public static bool TryParse(Type enumType, ReadOnlySpan<char> value, out object? result) =>
+          TryParse(enumType, value, ignoreCase: false, out result);
+
         public static bool TryParse(Type enumType, string? value, bool ignoreCase, out object? result) =>
             TryParse(enumType, value, ignoreCase, throwOnFailure: false, out result);
 
+        public static bool TryParse(Type enumType, ReadOnlySpan<char> value, bool ignoreCase, out object? result) =>
+            TryParse(enumType, value, ignoreCase, throwOnFailure: false, out result);
+
         private static bool TryParse(Type enumType, string? value, bool ignoreCase, bool throwOnFailure, out object? result)
+        {
+            if (value == null)
+            {
+                if (throwOnFailure)
+                {
+                    throw new ArgumentNullException(nameof(value));
+                }
+                result = null;
+                return false;
+            }
+
+            ReadOnlySpan<char> valueSpan = value.AsSpan();
+
+            return TryParse(enumType, valueSpan, ignoreCase, throwOnFailure, out result);
+        }
+
+        private static bool TryParse(Type enumType, ReadOnlySpan<char> value, bool ignoreCase, bool throwOnFailure, out object? result)
         {
             // Validation on the enum type itself.  Failures here are considered non-parsing failures
             // and thus always throw rather than returning false.
             RuntimeType rt = ValidateRuntimeType(enumType);
-
-            ReadOnlySpan<char> valueSpan = value.AsSpan().TrimStart();
-            if (valueSpan.Length == 0)
+            value = value.TrimStart();
```

Comment:
@terrajobst Just to double check, I suppose we want to keep TrimStart() here as opposed to the method above right? Difference is that if we keep it here then when calling the method with the TryParse overload that takes a span you will get trimmed too.

Reply:
@terrajobst Seems like we are only waiting for your answer :(

Reply:
@joperezr not sure I fully understand your question but I'd say all parsing methods should have the same semantics with respect to leading/trailing whitespace, regardless of whether they are `Parse` or `TryXxx`, take strings, or spans. Does that make sense?